tapestry.js

```diff
@@ -2199,34 +2299,47 @@ function getNodesDimensions(dataset) {
         // save max point so we can calculate our tapestry width and height
         if (dataset.nodes[index].fx > maxPointX) {
             maxPointX = dataset.nodes[index].fx;
+            if (maxPointX > getBrowserWidth() + MAX_RADIUS){
+                maxPointX = getBrowserWidth() + MAX_RADIUS;
+            }
         }
         if (dataset.nodes[index].fy > maxPointY) {
             maxPointY = dataset.nodes[index].fy;
+            if (maxPointX > getBrowserHeight() + MAX_RADIUS){
+                maxPointX = getBrowserHeight() + MAX_RADIUS;
+            }
         }
     }
-
+    
     return {
         'x': maxPointX + MAX_RADIUS,
         'y': maxPointY + MAX_RADIUS
     };
 }
 
 /* Gets the boundary of the tapestry */
-function getTapestryDimensions() {
+function getTapestryDimensions(ratio) {
```

Comment:
Please revert this function back to its original implementation unless you have a good reason why this change is needed. I reverted it and it worked well without all these changes.